User Status: Last seen field of full user profile and JS Tooltip Hovers for Buddy list and PM List.

vinitS101 opened the pull request:
Issue #11607

vinitS101 commented:
@rishig I think I've covered everything from #11607 . Please check it out.

rishig commented:
Awesome, thanks Vinit! A few things I noticed:
* This one is not yet done (which is fine, we can split that out into its own issue if you didn't intend to address it:

> If the user is Unavailable, we should show them as "Last online: today" to other users.
> Not sure the best way to do this, but ideally the backend would handle this.

* There's some regression for the conversation list on the left side (under "Private messages"). In particular, only the first availability circle is showing.
![image](https://user-images.githubusercontent.com/890911/53699150-29f9b780-3d9a-11e9-8d4a-247348fb4651.png)

* We should make these changes for the conversation list on the left side as well.

* Ideally these would be js hovers (that appear immediately) rather than html titles, but this is already a big improvement so we can merge with html titles and fix forward if necessary once the regression above is addressed.

vinitS101 commented:
Ah, I didn't notice the PM issue. I'll work on these as well as the "Last online: Today" message for Unavailable users.

vinitS101 commented:
@rishig I the last 3 commits I have:

- Fixed the regression where the PM List user circle's where not showing
- Unavailable user's now show as "Last online: today" to other users.
- Added similar hover text to the PM user circles Active/Idle/Offline/Unavailable for green/orange/white/white-with-line

I did this with the title html tags here because I felt like this was more consistent and in line with the other user circles (Plus they required very little changes). If you exclusively want js hovers here please let me know.

rishig commented:
Great, thanks! I think a lot of users actually mainly use the Private Messages list, so we do want the full information there. 

For the html vs js: we can leave it as html for now; we will want to move to js hovers, but that can be done as a separate effort.

vinitS101 commented:
I will definitely look into the JS hovers and see if I can do that later on (hopefully this weekend). Have a bunch of quizzes at school this week so I wanted to get this done before I started studying for those.

timabbott commented:
@vinitS101 there's a PR for JS hovers on reactions, see https://github.com/zulip/zulip/pull/10266, that we might want to share code wiht.

vinitS101 commented:
@timabbott I'll go through that. Can I make a new PR for that (later this week) or should I continue here?

vinitS101 commented:
Added JS tooltips instead of the HTML titles that we had before. Please let me know if there are any issues or if any changes are to be made.

vinitS101 commented:
Created a cleaner PR #11882
